test/ibmq/test_ibmq_job_attributes.py

```diff
@@ -349,6 +375,165 @@ def test_job_tags_and(self, provider):
+                         .format(job_id, job.tags()))
+
+    @requires_provider
+    def test_job_tags_add_and_remove(self, provider):
+        """Test updating job tags by adding and removing the same job tag."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+
+        job = backend.run(qobj, job_tags=[uuid.uuid4().hex], validate_qobj=True)
+        job_id = job.job_id()
+
+        replacement_tags = [uuid.uuid4().hex, uuid.uuid4().hex]
+        # Remove the first tag in `replacement_tags`
+        removal_tags = replacement_tags[:1]
+        # After updating, the final tags should only include the second tag of
+        # `replacement_tags`.
+        tags_after_update = replacement_tags[1:]
+
+        _ = job.update_tags(replacement_tags=replacement_tags, removal_tags=removal_tags)
+        # Cached results may be returned if updating the same job too quickly.
+        # Wait before updating again.
+        time.sleep(2)
+        job.refresh()
+        self.assertEqual(set(job.tags()), set(tags_after_update),
+                         'Updating the tags for job {} was unsuccessful.'
+                         'The tags are {}, but they should be {}.'
+                         .format(job_id, job.tags(), tags_after_update))
+
+    @requires_provider
+    def test_job_tags_all_parameters(self, provider):
+        """Test updating job tags by adding and removing the same job tag."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+
+        initial_job_tags = [uuid.uuid4().hex]
+        job = backend.run(qobj, job_tags=initial_job_tags, validate_qobj=True)
+        job_id = job.job_id()
+
+        replacement_tags = [uuid.uuid4().hex, uuid.uuid4().hex]
+        additional_tags = [uuid.uuid4().hex, uuid.uuid4().hex]
+        # Remove the first tag of `replacement_tags` and `additional_tags`.
+        removal_tags = replacement_tags[:1] + additional_tags[:1]
+        # After updating, the final tags should be a list containing the second tag
+        # of `replacement_tags` and `additional_tags`.
+        tags_after_update = replacement_tags[1:] + additional_tags[1:]
+
+        # Replace the tags with `new_tags` and remove the first new tag.
+        _ = job.update_tags(replacement_tags=replacement_tags,
+                            additional_tags=additional_tags,
+                            removal_tags=removal_tags)
+        # Cached results may be returned if updating the same job too quickly.
+        # Wait before updating again.
+        time.sleep(2)
+        job.refresh()
+        self.assertEqual(set(job.tags()), set(tags_after_update),
+                         'Updating the tags for job {} was unsuccessful.'
+                         'The tags are {}, but they should be {}.'
+                         .format(job_id, job.tags(), tags_after_update))
```

Comment:
Since updating & verifying job tags is repeated for 7 test cases, you can probably put it in a function to make the code more readable.